### PR TITLE
internal/contributebot: close prs created from a branch of the main repo instead of from a fork

### LINKED ISSUE #319: internal/contributebot: check for branches in shared repository
See the [design doc](https://github.com/google/go-cloud/blob/master/internal/contributebot/DESIGN.md#branches-in-fork-check) for this check.

(Continuation of #216)

### PR DESCRIPTION
**Pull request by vangent**
Fixes #319.

### CODE REVIEW

**Comment by vangent** [comment]
@zombiezen -- please feel free to reassign these reviews if you're okay with someone else looking at them. There are about 5 coming for various contributebot checks.

**Comment by zombiezen** [comment]
@shantuo, can you review?

File: internal/contributebot/main.go [hunk 1]
```diff
@@ -131,6 +133,33 @@ func (w *worker) receiveIssueEvent(ctx context.Context, e *github.IssuesEvent) e
...
+	pr, _, err := client.PullRequests.Get(ctx, owner, repo, prNumber)
+	if err != nil {
+		return err
+	}
+	data.PullRequest = pr
+
+	// Process the pull request, deciding what actions to take (if any).
+	edits := processPullRequestEvent(data)
+	// Execute the actions (if any).
+	return edits.Execute(ctx, client, owner, repo, prNumber)
```
Comment (shantuo) [inline internal/contributebot/main.go]: It seems to me that some arguments can be passed by a *PullRequestEvent and extract inside Execute, would that be clearer?

Comment (vangent) [reply internal/contributebot/main.go]: Now passing data to Execute as well, and xxxData struct has everything that Execute needs.

**Pull request merged**